### PR TITLE
Update sentry; fix config

### PR DESCRIPTION
**Pull request by zackkrida**
## Description

Updates the Sentry SDK and updates it to allow errors from openverse.org. Currently the Sentry SDK is way out of date on the frontend and not accepting events from openverse.org. This PR updates the SDK and fixed the allowed domains.

## Testing Instructions

I actually need help from @WordPress/openverse-frontend on the best way to test this.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/utils/sentry-config.ts [hunk 1]
```diff
@@ -15,8 +15,8 @@ export const sentryConfig: ModuleConfiguration = {
   logMockCalls: false,
   lazy: true,
   clientConfig: {
-    // Only allow errors that come from an actual openverse.engineering subdomain
-    allowUrls: [/^https?:\/\/(.*)\.openverse\.engineering/],
+    // Only allow errors that come from openverse.org or a subdomain
+    allowUrls: ["https://openverse.org", /^https?:\/\/(.*)\.openverse\.org/],
```
Comment (dhruvkb) [inline frontend/src/utils/sentry-config.ts:19]: The regex can include the first case.

```suggestion
    allowUrls: [/^https?:\/\/((.*)\.)?openverse\.org/],
```

**Review by dhruvkb — APPROVED** [review]

File: frontend/src/utils/sentry-config.ts (hunk 1 above)
Comment (krysal) [reply frontend/src/utils/sentry-config.ts:19]: According to https://regex101.com/, @dhruvkb's suggestion accepts the base URL plus strings like:
- https://staging.openverse.org
- https://api-staging.openverse.org
Is it something else missing?

Comment (zackkrida) [reply frontend/src/utils/sentry-config.ts:19]: 🤦 I had a typo in 'openverse' in one of my domains! Thanks for testing!

**Review by krysal — APPROVED** [review]
Good catch!

**Pull request merged**